test/test_torch.py

```diff
@@ -197,6 +197,48 @@ def test_sinh(self):
     def test_lgamma(self):
         self._testMathByName('lgamma')
 
+    def test_digamma(self):
+        makeshift = True  # TODO Replace makeshift digamma() with higher precision version.
+        x = torch.Tensor([-0.1, 3, 999])
+        expected = torch.Tensor([
+            9.24507305005294,
+            0.92278433509846,
+            6.90625419464,
+        ])
+
+        out = torch.digamma(x)
+        self.assertEqual(out, expected, 1e-5)
+
+        out = x.digamma()
+        self.assertEqual(out, expected, 1e-5)
+
+        if not makeshift:
+            y = torch.Tensor([-10, 0])
+            out = torch.digamma(y)
+            self.assertEqual(out == float('inf'), torch.ByteTensor([1, 1]))
+
+        # This tests against a scipy reference implementation.
+        x = torch.Tensor([
+            1.00000000e-05, 3.16227766e-05, 1.00000000e-04,
+            3.16227766e-04, 1.00000000e-03, 3.16227766e-03,
+            1.00000000e-02, 3.16227766e-02, 1.00000000e-01,
+            3.16227766e-01, 1.00000000e+00, 3.16227766e+00,
+            1.00000000e+01, 3.16227766e+01, 1.00000000e+02,
+            3.16227766e+02, 1.00000000e+03, 3.16227766e+03,
+            1.00000000e+04, 3.16227766e+04,
+        ])
```

Comment:
Can you compress that to `torch.linspace`?

Reply:
Done. (using `torch.logspace()`)